src/elisp/treemacs-tags.el

```diff
@@ -136,21 +136,25 @@ should be placed under."
   (let ((buff)
         (result)
         (mode)
+        (run-mode-hooks-override (lambda (&rest _) nil))
         (existing-buffer (get-file-buffer file)))
     (if existing-buffer
         (setq buff existing-buffer)
-      (setq buff (find-file-noselect file)))
-    (with-current-buffer buff
-      (setq result (condition-case _
-                       (imenu--make-index-alist t)
-                     (error nil))
-            mode major-mode))
-    (unless existing-buffer (kill-buffer buff))
-    (when result
-      (when (string= "*Rescan*" (car (car result)))
-        (setq result (cdr result)))
-      (unless (equal result '(nil))
-        (treemacs--post-process-index result mode)))))
+      (progn
+        (advice-add 'run-mode-hooks :override run-mode-hooks-override)
```

Comment:
That's a good case for `cl-flet`:
`(cl-flet ((run-mode-hooks (&rest _) (ignore))) ... )`